src/main/java/microsoft/exchange/webservices/data/property/complex/ComplexPropertyCollection.java

```diff
@@ -103,15 +103,13 @@ protected ComplexPropertyCollection() {
    *
    * @param complexProperty The complex property.
    */
-  protected void itemChanged(ComplexProperty complexProperty) {
-    EwsUtilities
-        .ewsAssert(complexProperty instanceof ComplexProperty, "ComplexPropertyCollection.ItemChanged",
-                   String.format("ComplexPropertyCollection." +
-                                 "ItemChanged: the type of " +
-                                 "the complexProperty argument " +
-                                 "(%s) is not supported.", complexProperty.getClass().getName()));
-
-    TComplexProperty property = (TComplexProperty) complexProperty;
+  protected void itemChanged(final ComplexProperty complexProperty) {
```

Comment:
Completely agree. I've added generic type parameter in the `IComplexPropertyChangedDelegate` interface (to keep strong contract) and remove few casts.